netweaver/extract_installer.sls

```diff
@@ -0,0 +1,22 @@
+{%- from "netweaver/map.jinja" import netweaver with context -%}
+{% set host = grains['host'] %}
+{% for node in netweaver.nodes if node.host == host and (netweaver.sapcar_exe_file is defined and netweaver.swpm_sar_file and netweaver.swpm_extract_dir is defined) %}
```

Comment:
We don't need this loop, right? It's quite useless. I would just remove it. The code will do the same without it. In fact, with the loop, the states will be executed needlessly multiple times, potentially creating issues.

Reply:
Yes, this state will execute on each host and we have the global extraction data which is common to all hosts, so its fine to remove it.

However, removing the loop, also takes away ability to set the state name based of `sid` and `instance` from node pillar entries. And we do not have those entries in the grains. Should I just use host in the state name, only issue will be when 1 host having multiple instances.

Reply:
Why do you need any `sid`, `instance` or `host` ? The reason why this items are added to the state names is just to avoid duplicated state names, only because you loop in a machine. If you don't need to loop (because the action must be performed directly in all nodes), you can have just the same name for the state.

Just put `extract_installer_file` and that's it. This works totally fine. We don't need to add not needed extra complexity.